addons/stock/models/stock_move.py

```diff
@@ -815,31 +809,84 @@ def onchange_date(self):
         if self.date_expected:
             self.date = self.date_expected
 
-    @api.onchange('move_line_nosuggest_ids')
-    def onchange_move_line_nosuggest_ids(self):
+    @api.onchange('move_line_ids', 'move_line_nosuggest_ids')
+    def onchange_move_line_ids(self):
         breaking_char = '\n'
-        move_lines_to_create = []
-        for move_line in self.move_line_nosuggest_ids:
+        if self.picking_type_id.show_reserved:
+            move_lines = self.move_line_ids
+        else:
+            move_lines = self.move_line_nosuggest_ids
+
+        for move_line in move_lines:
             # Look if the `lot_name` contains multiple values.
             if breaking_char in (move_line.lot_name or ''):
                 splitted_lines = move_line.lot_name.split(breaking_char)
                 splitted_lines = list(filter(lambda line: line, splitted_lines))
                 move_line.lot_name = splitted_lines[0]
-                # For each SN line, set move ine data...
```

Comment:
use list(filter(None, split_lines)), it's supposed to be [faster](https://stackoverflow.com/questions/3845423/remove-empty-strings-from-a-list-of-strings)